src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -942,13 +943,29 @@ public void ResetByteRunInterruptionOffsets(ObjectData nodeContents)
             }
         }
 
+        private static unsafe void SetIdentity(ObjectWriter objectWriter)
+        {
+            // emit compiler identity in comment section: .NET AOT {version}
+            byte[] runtimeVersionBytes = objectWriter._sb.Clear().Append(".NET AOT ")
+                .Append(Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion)
+                .Append('\0').UnderlyingArray;
+
+            objectWriter.SetSection(ObjectNodeSection.CommentSection);
+            fixed (byte* version = &runtimeVersionBytes[0])
+            {
+                objectWriter.EmitBytes((IntPtr)version, runtimeVersionBytes.Length);
```

Comment:
Can we just replace this with a Span overload?

Reply:
Most of the uses of this API could be spanified and we even have a TODO for it in this file (the code pre-dates Span). We can introduce a span overload now or it would also be fine to leave as-is since it's par for the course.

Reply:
> Can we just replace this with a Span overload?

Switched to EmitBlob which doesn't require pinning at least.